tests/feature_resolution_editions.js

```diff
@@ -0,0 +1,524 @@
+
+    test.end();
+})
+
+tape.test("feature resolution inheritance file to message", function(test) {
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023";
+    option features.json_format = LEGACY_BEST_EFFORT;
+    option features.(abc).d_e = deeply_nested_false;
+
+    message Message {
+        string string_val = 1;
+        string string_repeated = 2 [features.enum_type = CLOSED];
+    }`).root.resolveAll();
+
+    // Should flip enum_type from default setting, inherit from Message,
+    // and keep everything else
+    test.same(rootEditionsOverriden.lookup("Message").fields.stringRepeated._features, {
+        enum_type: 'CLOSED',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' }
+    })
+    
+    test.end();
+});
+
+tape.test("feature resolution inheritance message to field", function(test) {
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023";
+
+    message Message {
+        option features.json_format = LEGACY_BEST_EFFORT;
+        option features.(abc).d_e = deeply_nested_false;
+        string string_val = 1;
+        string string_repeated = 2 [features.enum_type = CLOSED];
+    }`).root.resolveAll();
+
+    // Should flip enum_type from default setting, inherit from Message,
+    // and keep everything else
+    test.same(rootEditionsOverriden.lookup("Message").fields.stringRepeated._features, {
+        enum_type: 'CLOSED',
+        field_presence: 'EXPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_false' }
+    })
+    
+    test.end();
+});
+
+tape.test("feature resolution inheritance message to nested message", function(test) {
+    var rootEditionsOverriden = protobuf.parse(`edition = "2023"; 
+    message Message {
+        option features.json_format = LEGACY_BEST_EFFORT;
+        option features.(abc).d_e = deeply_nested_false;
+        string string_val = 1;
+        string string_repeated = 2 [features.enum_type = CLOSED];
+
+    message Nested {
+        option features.(abc).d_e = deeply_nested_true;
+        option features.field_presence = IMPLICIT;
+        int64 count = 9;
+    }
+    }`).root.resolveAll();
+
+    test.same(rootEditionsOverriden.lookup("Message").lookup("Nested")._features, {
+        enum_type: 'OPEN',
+        field_presence: 'IMPLICIT',
+        json_format: 'LEGACY_BEST_EFFORT',
+        message_encoding: 'LENGTH_PREFIXED',
+        repeated_field_encoding: 'PACKED',
+        utf8_validation: 'VERIFY',
+        '(abc)': { d_e: 'deeply_nested_true' } 
+    });
+    test.end();
+});
+
+tape.test("feature resolution inheritance file to enums and enum values", function(test) {
```

Comment:
nit: I think this one should be enum -> enum value, with features on the enum to inherit.  You already have file->enum below